### PR TITLE
SMPP transport throttling should handle reconnects

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

File: vumi/transports/smpp/smpp_transport.py [hunk 1]
```diff
@@ -499,6 +504,12 @@ def _check_stop_throttling(self):
         """
         self._unthrottle_delayedCall = None
 
+        if not self.service.is_bound():
+            # We don't have a bound SMPP connection, so try again later.
+            log.msg("Can't check throttling while unbound, trying later.")
+            self.check_stop_throttling(self.get_static_config().throttle_delay)
```
Comment (justinvdm) [inline vumi/transports/smpp/smpp_transport.py]: Does this mean we check to see if we should stop throttling again after `throttle_delay` seconds if the service isn't bound?

Comment (justinvdm) [inline vumi/transports/smpp/smpp_transport.py]: If that is the case, maybe we should mention something about how it is used when we are not yet bound in [the config field's description](https://github.com/praekelt/vumi/blob/feature/issue-909-smpp-throttling-reconnect/vumi/transports/smpp/config.py#L20-L22)?

**Comment by justinvdm** [comment]
Left a minor comment, but looks good. Not sure if someone with more SMPP battle scars should take a look though.

**Comment by hodgestar** [comment]
:+1:

File: vumi/transports/smpp/smpp_transport.py (hunk 1 above)
Comment (jerith) [inline vumi/transports/smpp/smpp_transport.py]: This is only relevant during a reconnect (we can't be throttled until we've sent at least one message), and it has more or less the same behaviour as retrying a throttled message and getting another RTHROTTLED response.

**Pull request merged**